ui/raidboss/emulator/overrides/RaidEmulatorOverlayApiHook.js

```diff
@@ -33,11 +33,29 @@ export default class RaidEmulatorOverlayApiHook {
           // nextSignificantState is a bit inefficient but given that this isn't run every tick
           // we can afford to be a bit inefficient for readability's sake
           const combatantState = combatant.nextSignificantState(timestamp).toPluginState();
-          if (msg.ids && msg.ids.includes(parseInt(id, 16)))
-            combatants.push(combatantState);
-          else if (msg.names && msg.names.includes(tracker.combatants[id].name))
+          combatantState.ID = combatant.id;
+          combatantState.Name = combatant.name;
+          combatantState.Level = combatant.level;
+          combatantState.Job = combatant.job;
+          if ((msg.ids && msg.ids.length) || (msg.names && msg.names.length)) {
+            if (msg.ids && msg.ids.includes(parseInt(id, 16)))
+              combatants.push(combatantState);
+            else if (msg.names && msg.names.includes(tracker.combatants[id].name))
+              combatants.push(combatantState);
+          } else {
             combatants.push(combatantState);
+          }
         }
+        // @TODO: Move this to track properly
```

Comment:
What do you mean by this TODO? Move this where?  Can this be done properly in this PR, or just postponed for another PR (as I don't think triggers use these fields)?

Reply:
Oh, right! I forgot about those triggers.  How hard is it to track rather than try to derive them afterwards?  That seems like a much better solution.  Can it be done here?

Reply:
I was thinking of adding the tracking as a follow up PR after #2936 gets merged.

Reply:
Ok!